database/migrations/updates/add_id_to_attributes_in_revisions_table.php.stub

```diff
@@ -0,0 +1,42 @@
+<?php
+
+use Statamic\Eloquent\Database\BaseMigration as Migration;
+use Statamic\Eloquent\Revisions\RevisionModel;
+
+return new class extends Migration {
+    public function up()
+    {
+        // Extract the id from the key and add it to the attributes of the revision
+        RevisionModel::query()
+            ->whereJsonDoesntContainKey('attributes->id')
+            ->chunk(200, function ($revisions) {
+                foreach ($revisions as $revision) {
+                    $id = str($revision->key)
+                        ->afterLast('/')
+                        ->toInteger();
```

Comment:
This would break when not using integer ids for entries - can you change it?

Reply:
Ah yes, thanks for the notice! Will change it.